src/content/tutorials/user-auth-with-auth0/_index.md

```diff
@@ -0,0 +1,736 @@
+---
+title: 'User Auth with Auth0'
+showNew: true
+---
+
+In this tutorial, you'll integrate Auth0, an authentication and authorization provider, into a Cloudflare Workers application. Adding authorization and authentication to an application is a common task for developers, and by implementing it using Cloudflare Workers, you can take advantage of Workers' unique platform advantages to simplify how and when your application needs user data.
+
+## What you'll learn
+
+- How to authorize and authenticate users in Workers
+- How to persist authorization credentials inside of Workers KV
+- How to use Auth0 user info inside of your Workers application
+
+## Prerequisites
```

Comment:
(_My comment is regarding the whole **Prerequisites** section_)

```
## Prerequisites

To publish your Worker to Cloudflare, you'll need a few things:

- A Cloudflare account, and access to the API keys for that account
- An Auth0 account
- A Wrangler installation running locally on your machine, and access to the command-line

If you don't have those things quite yet, don't worry. We'll walk through each of them and make sure we're ready to go, before you start creating your application.

You'll need to get your Cloudflare API keys to deploy code to Cloudflare Workers: see ["Finding your Cloudflare API keys"](https://developers.cloudflare.com/workers/quickstart/#configure) for a brief guide on how to find them.
```

The title of “Prerequisites” feels wrong to me when it also says “If you don't have those things quite yet, don't worry. We'll walk through each of them and make sure we're ready to go, before you start creating your application.”

If it’s truly a prerequisite, then you should have done it before coming, and for each bullet point we should link you to the appropriate place for instructing you how to do those things.

I may have missed something, but afaict it seems that these bullets don’t link anywhere and the content instructing someone how to do these three bullets isn’t contained within the tutorial itself.

If it’s more like a prerequisite to the _meat_ of the tutorial, then I would rename the section something to the effect of “Setup” and link each bullet point to the section below, allowing a reader to skip it if they’ve already done those things. (I understand that this may have been a copy-pasta from other tutorials, but we should use this new tutorial as a model for what we should be doing, and update the prior tutorials if we have a better pattern.)

Finally, the section about getting Cloudflare API keys should be listed in the bulleted list just like the other necessary steps.

------

What I would recommend we do for tutorials in general is that we consolidate this stuff into a separate _tutorial_, called something like “Set up Cloudflare Workers” and we can start tutorials like this one with a link to that.

Reply:
thanks, this is really good feedback (and a written out version of what i think you sent me in chat before i went on pto, thanks for taking the time to write this out)

the quick start is designed to solve this exact thing, so i'm going to come back and update this section and:

1) require devs to go through the quick start before starting this tutorial
2) make "prerequisites" entirely focus on _external_ stuff you need to figure out: in this case, your auth0 account 

i'll probably still call out specific sections of the quick start later in the tutorial (setting up domains, etc), but copy-pasting this from other tutorials feels like an anti-pattern so will def improve on this. ty!

Reply:
@adamschwartz took a stab at improving this section - https://github.com/cloudflare/workers-docs/pull/726/commits/1c4736d571f656cb57ab03f288ac097f0232356c - let me know what you think!